### PR TITLE
feat: Disable motion

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref https://github.com/ant-design/ant-design/discussions/40040

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    ConfigProvider `theme` support `motion` to disable all component motion.       |
| 🇨🇳 Chinese |    ConfigProvider `theme` 支持 `motion` 配置以关闭全部动画过渡效果。      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 63ad334</samp>

This pull request enhances and fixes some components of the ant-design library, such as `Button`, `AutoComplete`, `Cascader`, and `ConfigProvider`. It also adds some new features and components, such as `ThemeSwitch`, `IconWrapper`, and `InlinePopover`, to improve the user experience and functionality of the documentation site. It also updates the documentation and tests to reflect the changes.

### <samp>🤖 Generated by Copilot at 63ad334</samp>

*  Add a new component `InlinePopover` for showing preview images when hovering over links ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e204a244fb8aea567f0534be3eee4bab3ecd176ef2e11c22c0dae861d5da746cR1-R50))
*  Add a new prop `motion` to the `ThemeSwitch` and `App` components, and a new state variable `isMotionOff` to the `ThemeSwitch` component, for toggling the motion effect of the documentation site ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447R21-R23), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447R59-R74), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-5d7076c9ed6d07433cbe0e9ec9d4940ec94b53422e0475998ac2ce2e9caa3b46L44-R47), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-5d7076c9ed6d07433cbe0e9ec9d4940ec94b53422e0475998ac2ce2e9caa3b46R119))
*  Add new locale messages for the `ThemeSwitch` component in English and Chinese ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-77dff52935aad07ad5a5a47203c59f4d583e74034fd4695d189bd0c632b86e4bR6-R7), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-b823a78cf4beb8e8ede2c847a6fdfab8b61870bf9deffb35ec6fb7133384e281R6-R7))
*  Add new props `dropdownMatchSelectWidth` and `popupMatchSelectWidth` to the `AutoComplete` component, for controlling the width of the dropdown menu and the popup menu ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-c55203058d3a4064d8f52a5b4e7d7770153d215966d276d6814011d99268f76dR48-R50))
*  Add new props `classNames` and `styles` to the `Button` component, for customizing the semantic DOM elements of the button ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140R43-R44), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140R104), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140R114), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L231-R238))
*  Add a new component `IconWrapper` for wrapping the icon element of the `Button` component and applying custom styles and classes to it ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-d26216d96fb33d65170cc33ea1ca12f6f0be8c35b2f2fc304309a814f078709fR1-R23), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140R17), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L231-R238), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-2107ebf32abbf181377ff77c8318c01ab12c098f9276fa74c6d8ea4956f3c33dL23-R56), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-2107ebf32abbf181377ff77c8318c01ab12c098f9276fa74c6d8ea4956f3c33dL47-R85))
*  Add new entries and subsections to the `API` section of the `Button` component documentation in English and Chinese, for explaining the usage of the new props ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-80ce0ce5862cdbde65634fdeed1081c7812fabe85b736155e1be0c5da5760d16R58), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-80ce0ce5862cdbde65634fdeed1081c7812fabe85b736155e1be0c5da5760d16R68), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-80ce0ce5862cdbde65634fdeed1081c7812fabe85b736155e1be0c5da5760d16R75-R80), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-447d93c86c05c00f9924d883b65877d29b1d52a00560746e69352a1d1c8af4ffR63), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-447d93c86c05c00f9924d883b65877d29b1d52a00560746e69352a1d1c8af4ffR73), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-447d93c86c05c00f9924d883b65877d29b1d52a00560746e69352a1d1c8af4ffR80-R85))
*  Add new test cases for the `Button` component with the new props ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-276fe24dfac9bfa7de4a29f91e72acd80a99b1034b459bc7b7ebf2db5c47805cR164-R178))
*  Add new style rules and tokens for the `Button` component, for adjusting the margin and font size of the icon element ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L13-R15), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L44-R56), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L430-R442), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693R476), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693R487), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693R511))
*  Add a new static method `generateCalendar` to the `Calendar` component, for creating a custom calendar component with a different date library ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-0a5821c1b32fdf9bf1d3bb55702937aead9206b6762df0497ff7b030f405a21aR541-R546), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-013735da2a9e27cd090e2a7b89864881fcab5e33ff74b538079f42b82ccd3d24L8-R15))
*  Replace the `virtual` and `dropdownMatchSelectWidth` props from the `ConfigContext` with the `popupOverflow` prop, for controlling the popup logic of the `Cascader` component ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L164-R164), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L279-R278))
*  Add new props `popupMatchSelectWidth` and `popupOverflow` to the `ConfigProvider` component, for controlling the popup behavior of the select-like components ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR37-R38), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaL61-R64), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563L58-R59))
*  Add new test cases for the `ConfigProvider` component with the new props ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-d2ae73049106961068a29073558e18c4089b8060a46ae201d5ba8eb91d3a2fa2R1-R111))
*  Remove a test case for the `ConfigProvider` component with the deprecated props ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-8e729604931fbfd516e8911f74c3cca0529812b4c15f9c886d09f3ab663b367fL116-L124))
*  Reorder some imports of antd components and hooks in some files ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447L1-R10), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-5d7076c9ed6d07433cbe0e9ec9d4940ec94b53422e0475998ac2ce2e9caa3b46L8-R12), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L2-R7))
*  Remove an unused import of `useLayoutState` from the `GlobalLayout` component ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-5d7076c9ed6d07433cbe0e9ec9d4940ec94b53422e0475998ac2ce2e9caa3b46L17))
*  Remove some duplicated imports of antd utils and types from the `AutoComplete` component ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-c55203058d3a4064d8f52a5b4e7d7770153d215966d276d6814011d99268f76dL23-L26))
*  Remove a redundant rule from the `genSizeButtonStyle` function ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L444-L447))
*  Fix a mistake of removing the `iconCls` parameter from the `genSizeButtonStyle` function ([link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L400), [link](https://github.com/ant-design/ant-design/pull/41856/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693R417))

### CODE REVIEW

File: components/theme/util/alias.ts [hunk 1]
```diff
@@ -31,6 +31,14 @@ export default function formatToken(derivativeToken: RawMergedToken): AliasToken
   const screenXL = 1200;
   const screenXXL = 1600;
 
+  // Motion
+  if (mergedToken.motion === false) {
+    const fastDuration = '0.0001s';
```
Comment (afc163) [inline components/theme/util/alias.ts:36]: 为啥不是 0s ？

Comment (zombieJ) [reply components/theme/util/alias.ts:36]: 和 CSSMotion 对齐。CSSMotion 里设置 skip 后切换是按帧计算。这边 hover 态原生 transition 也做成晚一帧。

Comment (afc163) [reply components/theme/util/alias.ts:36]: 这个相当于 antd 要感知 CSSMotion 的内部逻辑，有点坏味道。

Comment (zombieJ) [reply components/theme/util/alias.ts:36]: 嗯，那先改 0 了。有问题再修。

**Comment by li-jia-nan** [comment]
Conflicting

**Review by li-jia-nan — APPROVED** [review]

**Comment by MadCcc** [comment]
https://preview-41856-ant-design.surge.sh/components/segmented-cn?theme=motion-off
Segmented 会有问题

**Comment by MadCcc** [comment]
https://user-images.githubusercontent.com/27722486/233025951-c6786592-da8f-4360-b90c-21da0cd4e8ca.mp4

Switch 会稍微有点怪

**Comment by afc163** [comment]
可以在 https://ant.design/docs/react/customize-theme 这里加一段文档说明。

**Pull request merged**